shotover-proxy/benches/cassandra_benches.rs

```diff
@@ -105,50 +134,64 @@ fn cassandra(c: &mut Criterion) {
+            resources
+                .connection
+                .execute(&stmt!(
+                    "CREATE TABLE test_protect_keyspace.test_table (pk varchar PRIMARY KEY, cluster varchar, col1 varchar, col2 int, col3 boolean);"
+                ))
+                .wait()
+                .unwrap();
+            resources
+                .connection
+                .execute(&stmt!(
+                    "INSERT INTO test_protect_keyspace.test_table (pk, cluster, col1, col2, col3) VALUES ('pk1', 'cluster', 'Initial value', 42, true);"
+                ))
+                .wait()
+                .unwrap();
+
+            resources
+        });
         for query in queries {
             // Benches the case where the message meets the criteria for encryption
             group.bench_with_input(
                 format!("protect_local_{}_encrypted", query.name),
-                || {
-                    let resources = BenchResources::new(
-                        "example-configs/cassandra-protect-local/topology.yaml",
-                        "example-configs/cassandra-protect-local/docker-compose.yml"
-                    );
-
-                    resources
-                        .connection
-                        .execute(&stmt!(
-                            "CREATE KEYSPACE test_protect_keyspace WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };"
-                        ))
-                        .wait()
-                        .unwrap();
-                    resources
-                        .connection
-                        .execute(&stmt!(
-                            "CREATE TABLE test_protect_keyspace.test_table (pk varchar PRIMARY KEY, cluster varchar, col1 varchar, col2 int, col3 boolean);"
-                        ))
-                        .wait()
-                        .unwrap();
-                    resources
-                        .connection
-                        .execute(&stmt!(
-                            "INSERT INTO test_protect_keyspace.test_table (pk, cluster, col1, col2, col3) VALUES ('pk1', 'cluster', 'Initial value', 42, true);"
-                        ))
-                        .wait()
-                        .unwrap();
-
-                    resources
-                },
-                |b, state| {
+                &resources,
+                |b, resources| {
                     b.iter(|| {
-                        state.connection.execute(&query.statement).wait().unwrap();
+                        let mut resources = resources.borrow_mut();
+                        let connection = &mut resources.as_mut().unwrap().connection;
+                        connection.execute(&query.statement).wait().unwrap();
                     })
                 },
             );
         }
     }
 }
 
+// This can be easily pulled out and reused by other benchmarks if needed.
+fn new_lazy_shared<T>(create: impl Fn() -> T) -> impl Fn() -> Arc<RefCell<Option<T>>> {
```

Comment:
Mind adding a comment how this works?

Reply:
yeah, this function touches a whole bunch of weird corners of rust.
I added some comments explaining what it does, let me know if you have any questions.

Reply:
Yep there is a bit of magic in there. Thanks!